apps/web/stores/AnswerStore.ts

```diff
@@ -2,28 +2,116 @@
 import { makeAutoObservable } from "mobx";
 // local
 import { WalkthroughRootStore } from "./WalkthroughRootStore";
+import {
+  QuestionVariableData,
+  VariableToSetPropertyName,
+} from "@repo/data/useWalkthroughData";
+import {
+  answerValuesAreNotEqual,
+  getVariableItemValue,
+} from "../utils/logic/variableItem";
+import { isObject, isString } from "../utils/typeChecking";
 
-export type WalkthroughAnswerType = string | string[] | Record<string, string>;
-export type WalkthroughStoreAnswers = Record<string, WalkthroughAnswerType>;
+export type AnswerTypes = string | string[] | Record<string, string>;
+export type AnswerState = Record<string, AnswerTypes>;
+export type AnswerToCheckValueFn = (
+  answerToCheck: string,
+) => AnswerTypes | undefined;
 
 export const DEFAULT_ANSWER_VALUE_MULTI_CHOICE = undefined;
-export const DEFAULT_ANSWER_VALUE_MULTI_CHOICE_MULTI = [];
+export const DEFAULT_ANSWER_VALUE_MULTI_CHOICE_MULTI: string[] = [];
 
 export class AnswerStore {
   rootStore: WalkthroughRootStore;
 
-  answers: WalkthroughStoreAnswers = {};
+  answers: AnswerState = {};
 
   constructor(rootStore: WalkthroughRootStore) {
     makeAutoObservable(this);
     this.rootStore = rootStore;
   }
 
-  setAnswerValue = (value: WalkthroughAnswerType, questionId: string) => {
+  handleVariableItem = (
+    currentItem: QuestionVariableData,
+    currentId: string,
+  ) => {
+    // get state references
+    const {
+      navigationStore: { handleForwardNavigation, addItemIdToHistory },
+    } = this.rootStore;
+
+    // get new variable value
+    const newItemValue = getVariableItemValue(
+      currentItem[VariableToSetPropertyName],
+      this.getAnswerToCheckValue,
+    );
+
+    // get current variable value
+    const currentItemValue = this.answers[currentId];
+
+    // compare with current variable value to see if it's changed
+    // if changed, update variable value and erase future state
+    if (
+      !currentItemValue ||
+      (currentItemValue &&
+        answerValuesAreNotEqual(currentItemValue, newItemValue))
+    ) {
+      // erase future state
+      this.eraseAnswersAfterCurrentItem();
+
+      // set answer value
+      this.answers[currentId] = newItemValue;
```

Comment:
This goes for a lot of these comments. I feel they are redundant. I think you've done a good job naming functions/methods as well so things like this below is also redundant.
```
// add variable item to history
 addItemIdToHistory(currentId);
```

Reply:
Refactored across the board! LMK what you think.